File: lizard-web/api/models/hide.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional


class HideItem(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: int
    hide_num: str
    processed_at: str
    class_name: str = Field(serialization_alias="class")
    yield_pct: Optional[float] = None
    area: Optional[float] = None
    op: str


class HideListResponse(BaseModel):
    items: list[HideItem]
    total: int
    page: int
    limit: int


class HideDetail(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: int
    hide_num: str
    op: str
    processed_at: str
    class_name: str = Field(serialization_alias="class")
    yield_pct: Optional[float] = None
    area: Optional[float] = None
    image_available: bool = False
    prev_hide_id: Optional[int] = None
    next_hide_id: Optional[int] = None

File: lizard-web/frontend/src/types/hide.ts
export interface HideItem {
  id: number
  hide_num: string
  processed_at: string
  class: string
  yield_pct: number | null
  area: number | null
  op: string
}

export interface HideListResponse {
  items: HideItem[]
  total: number
  page: number
  limit: number
}

export interface Hide {
  id: number
  hide_num: string
  op: string
  processed_at: string
  class: string
  yield_pct: number | null
  area: number | null
  image_available: boolean
  prev_hide_id: number | null
  next_hide_id: number | null
}

export type HideDetail = Hide
